src/Octoshift/GithubApi.cs

```diff
@@ -459,5 +460,145 @@ public virtual async Task<string> GetArchiveMigrationUrl(string org, int migrati
+                    {
+                        claimant = new Claimant
+                        {
+                            Id = (string)mannequin["claimant"]["id"],
+                            Login = (string)mannequin["claimant"]["login"]
+                        };
+                    }
+
+                    return new Mannequin
+                    {
+                        Id = (string)mannequin["id"],
+                        Login = (string)mannequin["login"],
+                        MappedUser = claimant
+
+                    };
+                }
+
+                payload = BuildListMannequinPayload(orgid, (string)data["data"]["node"]["mannequins"]["pageInfo"]["endCursor"]);
+
+            } while ((bool)data["data"]["node"]["mannequins"]["pageInfo"]["hasNextPage"]);
+
+            return new Mannequin();
+        }
+
+        public virtual async Task<string> GetUserId(string login)
+        {
+            var url = $"{_apiUrl}/graphql";
+
+            var payload = new
+            {
+                query = "query($login: String!) {user(login: $login) { id, name } }",
+                variables = new { login }
+            };
+
+            var response = await _client.PostAsync(url, payload);
+            var data = JObject.Parse(response);
+
+            return data["data"]["user"].Any() ? (string)data["data"]["user"]["id"] : null;
+        }
+
+        public virtual async Task<bool> ReclaimMannequin(string orgId, string mannequinId, string targetUserId)
+        {
+            var url = $"{_apiUrl}/graphql";
+
+            var payload = BuildReclaimMannequinPayload(orgId, mannequinId, targetUserId);
+
+            var response = await _client.PostAsync(url, payload);
+            var data = JObject.Parse(response);
+
+            return data["data"]["createAttributionInvitation"].Any()
+                && (string)data["data"]["createAttributionInvitation"]["source"]["id"] == mannequinId
+                && (string)data["data"]["createAttributionInvitation"]["target"]["id"] == targetUserId;
+        }
+
+        private object BuildListMannequinPayload(string orgId, string cursor)
+        {
+            return new
+            {
+                query = @"query ($id: ID!, $after: String) {
+	                node(id: $id) {
+		                ... on Organization {
+			                mannequins(first: 100, after: $after) {
+				                pageInfo {
+					                endCursor
+					                hasNextPage
+				                }
+				                totalCount
+				                nodes {
+					                login
+					                id
+					                claimant {
+						                login
+						                id
+					                }
+				                }
+			                }
+		                }
+	                }
+                }
+                ",
+                variables = new { after = cursor, id = orgId }
+            };
+        }
```

Comment:
instead of building the query like this, it's better to follow our existing pattern which is to create the `query` part as a string constant and then add variables, like the way you did in `GetUserId`.

Reply:
I extracted the query to a method for 2 reasons. To make the method smaller since the query is too big and makes the method hard to read and also because of the pagination.

For example in `GetUserId` the query is set from an anonymous type, so the variable is immutable and you can't just set `after` on the variables, you need to _recreate_ the query, and that is the second reason why this was extracted as a method.

In the `GetMannequin` method this is used twice

# 468
```csharp
 var payload = BuildListMannequinPayload(orgid, null);
```
and in 502
```csharp
payload = BuildListMannequinPayload(orgid, (string)data["data"]["node"]["mannequins"]["pageInfo"]["endCursor"]);
```

I don´t follow this commend
> ... to create the query part as a string constant and then add variables, like the way you did in GetUserId.

because the method does separate query and variables

Reply:
refactored the code to use refactored to use `PostGraphQLWithPaginationAsync` in [d48c3](https://github.com/github/gh-gei/pull/271/commits/d48c3bfd20422b7a65ac851b6232341c29f88a3e)